mypy/messages.py

```diff
@@ -1842,25 +1857,32 @@ def format_literal_value(typ: LiteralType) -> str:
         return 'object'
 
 
-def collect_all_instances(t: Type) -> List[Instance]:
-    """Return all instances that `t` contains (including `t`).
+def collect_all_names(t: Type) -> List[Tuple[str, str]]:
+    """Return all (shortname, fullname) pairs for all types that show
+    up when printing `t` in an error message.
```

Comment:
This docstring is misleading. This function only collects "named" types, like instances and type aliases.

Reply:
I updated the docstring. Could you take a look and see if it's clearer?

Eventually, we probably should update this to include the names of all types that show up when printing the type in an error message, but that seems like something for a different PR.